packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTreeComponent.tsx

```diff
@@ -4,171 +4,255 @@
 *--------------------------------------------------------------------------------------------*/
 
 import "./ModelsTree.scss";
-import React, { useCallback, useEffect, useState } from "react";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
 import { useActiveIModelConnection, useActiveViewport } from "@itwin/appui-react";
 import { GeometricModel3dProps, ModelQueryParams } from "@itwin/core-common";
-import { IModelApp, Viewport } from "@itwin/core-frontend";
-import { ModelTreeProps } from "../../../types";
+import { IModelApp, IModelConnection, ScreenViewport, Viewport } from "@itwin/core-frontend";
+import { ModelsTreeHeaderButtonProps, ModelTreeProps } from "../../../types";
 import { TreeHeaderComponent } from "../../header/TreeHeader";
 import { useTreeFilteringState } from "../../TreeFilteringState";
 import { AutoSizer } from "../../utils/AutoSizer";
 import { toggleAllCategories } from "../CategoriesVisibilityUtils";
 import { ModelsTree } from "./ModelsTree";
+import { IconButton } from "../../IconButton";
+import { TreeWidget } from "../../../TreeWidget";
 
-interface TreeViewModelInfo {
-  id: string;
-  isPlanProjection?: boolean;
-}
+export namespace ModelsTreeComponentNamespace {
```

Comment:
I would suggest instead of using `namespace` put buttons on `ModelsTreeComponent`:

```typescript
export function ModelsTreeComponent(props: ModelTreeProps) {
  // models tree component implementation
}

ModelsTreeComponent.ShowAllButton = ShowAllButton;
ModelsTreeComponent.HideAllButton = HideAllButton;
ModelsTreeComponent.InvertButton = InvertButton;
ModelsTreeComponent.View2DButton = View2DButton;
ModelsTreeComponent.View3DButton = View3DButton;
```

And then there is no need to export each button component itself. Same for `CategoriesTreeComponent`